### ISSUE TITLE
Misplacement when pasting too many blocks

### ISSUE THREAD

**Issue by Henry0811** [maruohon/litematica#508]
* Minecraft version: 1.18.1
* malilib 0.11.8 and litematica 0.10.4
* When i'm playing Minecraft servers and try to paste but when the building is too big then the paste will misplace some blocks, these blocks are always some block containers like hopper and chest and also blocks with special uses like bed.

**Comment by maruohon** [comment]
That's too vague of a description for me to have any idea what actually happens. I assume you mean the creative mode `Paste Schematic in World` feature, when used on a server so that it works in the command-based mode?

Can you send a screenshot of an example case where the blocks got misplaced, so that I can see what/where/how was placed wrongly. A video of it happening would be even better. Also a screenshot of the Generic config category with the search term `past|comm` would be helpful (that's a pipe character in the middle for an "OR" operation).

**Comment by Henry0811** [comment]
OK and the misplaced blocks are sometimes pasted as "ghost block" which sometimes can't be seen by other online players.

**Comment by Henry0811** [comment]
![image](https://user-images.githubusercontent.com/60054887/160512783-9ead0712-83bc-4f59-a05d-a5e2a8e34146.png)
hmm that's the picture of the pasted emerald farm and you can see these things here.

**Comment by Henry0811** [comment]
Just every time when you're actually "in" the schematic and pasting big buildings you can find out that these blocks are pasted here.

**Comment by Henry0811** [comment]
And it is always pasting these blocks under the player's feet.

**Comment by maruohon** [comment]
These are the temporary blocks used for the NBT restore option. These should get removed immediately as well. Are you using too high of a command rate? Usually you need to lower it down quite a lot when you use the NBT restore option. But even so if these get let behind, are you on a Spigot/Paper based server that just drops packets if you send packets too fast? Try lowering the Generic -> `commandLimitPerTick` option to something like 8 or 4. And if you don't actually need to restore the NBT data, then disable the NBT restore option, as it has a LOT of extra packet overhead over normal pasting.

**Comment by Henry0811** [comment]
oh then i have another question
why sometimes when you saving a schematic, it will replace some of the blocks to "unknown blocks" which can only replaced by /fill and /setblock command.

**Comment by maruohon** [comment]
When saving? What do you mean? The schematic has "unknown blocks"? The only case that I know of that is problematic, is if you save a running redstone contraption, then you will get the moving piston heads (and powered observers) in the schematic, and if you paste that on a server, then you will get the non-functional piston heads without their full block entity data, and other blocks like powered observers without their scheduled block ticks.

Basically don't save running redstone contraptions, stop them before saving a schematic of it.

**Comment by Henry0811** [comment]
no
I mean the "unknown blocks" just look like air but you can't place anything on it and you can only replace it by /fill or /setblock command, you can't just destroy them in a normal way, but you can walk through it. It just has every characteristics of air except one——you cannot place blocks on it

**Comment by Henry0811** [comment]
Oh and yeah i think when i'm saving that piston is actually moving

**Comment by Henry0811** [comment]
And you're right the "unknown blocks" is the running pistons but it is not saved as the piston head or something.

**Comment by maruohon** [comment]
It **is** saved as a piston head. But when you paste it in multiplayer, the mod can't restore the proper full block entity data for it, and a moving piston head without the block entity data is invisible, the only way you know it's there is *because you can't place blocks into that position*.

**Comment by Henry0811** [comment]
Ok thanks
